#158755849 Static page management

OlatunjiYso opened the pull request:
#### What does this PR do?
This PR allows the admin to manage static pages and users view static pages
#### Description of Task to be completed?
- add model for storing static pages in database
- create plugin for static pages
- add templates for viewing static page (users)
- add templates for static page management (admin)
- add components and containers for static pages
- add forms to add and edit static page
- add table for managing static pages
#### How should this be manually tested?
- pull the branch
- run `reaction reset -n`
- reaction to start the app
- signin as admin and find the pages icon on the admin side-nav
- select it and add and manage pages
#### Screenshots (if appropriate)
<img width="1440" alt="screen shot 2018-07-19 at 12 28 41 pm" src="https://user-images.githubusercontent.com/27797745/42940607-06dd0ed2-8b52-11e8-98c8-0ff9fdecbf26.png">
<img width="1104" alt="screen shot 2018-07-19 at 12 24 49 pm" src="https://user-images.githubusercontent.com/27797745/42940748-7c6e5f70-8b52-11e8-99e6-feef61cf393d.png">
<img width="1440" alt="screen shot 2018-07-19 at 12 27 17 pm" src="https://user-images.githubusercontent.com/27797745/42940619-1082961e-8b52-11e8-8a97-310d3cff8919.png">
<img width="1350" alt="screen shot 2018-07-19 at 12 26 47 pm" src="https://user-images.githubusercontent.com/27797745/42940636-1c8c2556-8b52-11e8-9877-c0fef75d05ab.png">


#### What are the relevant pivotal tracker stories?
[Finishes #158755849](https://www.pivotaltracker.com/n/projects/2182264)

#### Any background context you want to provide?
N/A

ddouglasz commented:
LGTM 😋. At least I came around to your desk to see what you were doing. Good job bro.

saheedt commented:
@Daramola98 and I tested this, look good to go. Great job @OlatunjiYso

abdulfataiaka commented:
@OlatunjiYso, Is there a reason why you are changing core plugins instead of creating a new plugin for static pages?

OlatunjiYso commented:
@abdulfataiaka , I created my react components within the core plugin because I found other other react components there.

Rhotimee commented:
Works fine. LGTM

abdulfataiaka commented:
@OlatunjiYso, your `staticPage` component file was placed in the `accounts` plugin folder, which was meant to handle `account features` only and does not have anything to do with static pages, the plugin for the static page feature should be in the `/imports/plugins/custom` folder same way the `vendor-tour` plugin created by @Daramola98 is, and plugins are not in the core folder because they have react components, its because they introduce core feature into `reactioncommerce`.

abdulfataiaka commented:
@OlatunjiYso 

- The scope of your commit message should be pascal cased and not spaced
- You commit message should have a very short, precise and descriptive title
- The commit message body should hold the listing of what have been done to implement the feature preceded with -the `-` character

Please checkout the link below
https://github.com/andela/asgard-rc/wiki/Git-naming-conventions-and-best-practices

abdulfataiaka commented:
@OlatunjiYso, the storyId in your PR title should not be in this => `[    ]`, and the first letter of your PR title should be capitalized as in a `sentence`

OlatunjiYso committed:
ft(staticPages): implement static page management
- add static page schema
- add forms to add and update static page
- add static page plugins
- rename folders to follow convention

[Delivers #158755849]

OlatunjiYso commented:
@abdulfataiaka , I have implemented the feedback given.

abdulfataiaka commented:
@OlatunjiYso, please pay attention to the convections, you seem to be missing out on some things, in your branch name
`ft-static-pages-#158755849` should be `ft-static-pages-158755849`
And in your commit message
should start with `feat` not `ft`

abdulfataiaka commented:
@OlatunjiYso, notice that plugins folder names are not `camel` or `pascal` cased, please rectify this.

abdulfataiaka commented:
@OlatunjiYso, lastly, why do we have three separate plugins for same feature, why can't we have every functionalities of the static page scoped under on plugin folder?

clintfidel commented on imports/plugins/included/default-theme/client/styles/staticPages.less:
```diff
@@ -0,0 +1,17 @@
+
+.pageManager {
+    margin-top: 4rem;
+}
```
@OlatunjiYso , I think you should look into the naming convention for styles, you using pascalCase and snake-case, follow a convention a use it all through.

clintfidel reviewed (commented):
Good one @OlatunjiYso , please look into the comment I dropped, thanks

clintfidel commented:
@OlatunjiYso , I would suggest that you follow color scheme while designing the template for static pages as the colors you used seems off, please look into the color scheme. Thanks

OlatunjiYso commented:
@abdulfataiaka , the names of the static page folders are only similar, they are very much different.
As such, I may not be able to scope them into a single folder. 

- The first folder `static-page` is for viewing the static page
- The second folder `static-page-management` is for the admin who adds, edit and delete static pages
- The last folder `static-page-component` is for the nav.

Putting all these together will require me declare a common template, route and name for the three which I believe wont work out.
What do you think about this?

abdulfataiaka commented:
@OlatunjiYso, you have made a good point, so please address @clintfidel's comments.

abdulfataiaka commented:
@OlatunjiYso, this is what I see when I tried it out 👇 

<img width="476" alt="screen shot 2018-07-19 at 2 38 42 pm" src="https://user-images.githubusercontent.com/32123313/42997320-2f6f7b16-8c0e-11e8-8be6-61947c365b61.png">
<img width="476" alt="screen shot 2018-07-19 at 2 39 14 pm" src="https://user-images.githubusercontent.com/32123313/42997321-2f926fcc-8c0e-11e8-91e4-25e603ac4c5a.png">

abdulfataiaka commented:
@OlatunjiYso, you seem to have not addressed the branch name and commit message comments

OlatunjiYso commented:
@ddouglasz 
@saheedt 
@abdulfataiaka 
@clintfidel 
@Daramola98 
@Rhotimee 
@abdulfataiaka 
Please I may need to bring down this PR because,  implementing the TTL's feedback requires me to rename the branch.
I have renamed the branch and addresses other feedback too. I will be raising a new PR off that branch.
The PR will be up shortly, please bear with me.

OlatunjiYso commented:
Please lets continue the conversation in the new PR
